src/main/Api.ts

```diff
@@ -1,16 +1,19 @@
 import { AxiosInstance } from 'axios';
 import { User } from './interfaces/User';
+import { Logger } from './interfaces/Logger';
 
 export class Api {
   constructor(
-    private readonly axios: AxiosInstance
+    private readonly axios: AxiosInstance,
+    private readonly logger: Logger
   ) { }
 
   public getUsersByEmail(email: string): Promise<User[]> {
     return this.axios
       .get('/api/v1/users', { params: { 'query': 'email:' + email } })
       .then(results => results.data)
-      .catch(err => {
+      .catch(error => {
+        this.logger.error(`${error.stack || error}`);
```

Comment:
I just realised I forgot to put a comment about logging this on the initial PR hah. Nice catch!

Reply:
Yeah I deliberately left it out of the initial PR to find out how to do logging properly in nodejs...nothing escape your eyes!! :)